m4 does not build with clang

``` bash
spack install m4%clang
```
produces a linker error
```
make[2]: Entering directory `/spack-stage/spack-stage-1tj6uI/m4-1.4.18/spack-build/src'
  CC       m4.o
  CC       builtin.o
  CC       debug.o
  CC       eval.o
  CC       format.o
  CC       freeze.o
  CC       macro.o
  CC       input.o
  CC       output.o
  CC       path.o
  CC       symtab.o
  CCLD     m4
../lib/libm4.a(xmalloc.o): In function `xnmalloc':
spack/var/spack/stage/m4-1.4.18-tdbq4h34ahmclwg3cyzlpg4rnzcefsoz/m4-1.4.18/lib/xmalloc.c:(.text+0x71): undefined reference to `__muloti4'
```

I suspect the build fails to find symbols that otherwise come from libgcc. I tried setting ```LIBS=-lcompiler-rt``` without success.

--- a/var/spack/repos/builtin/packages/m4/package.py
+++ b/var/spack/repos/builtin/packages/m4/package.py
@@ -48,7 +48,7 @@ class M4(AutotoolsPackage):
         spec = self.spec
         args = ['--enable-c++']
 
-        if spec.satisfies('%clang'):
+        if spec.satisfies('%clang') and not spec.satisfies('platform=darwin'):
             args.append('CFLAGS=-rtlib=compiler-rt')
 
         if '+sigsegv' in spec:

--- a/var/spack/repos/builtin/packages/m4/package.py
+++ b/var/spack/repos/builtin/packages/m4/package.py
@@ -48,6 +48,9 @@ class M4(AutotoolsPackage):
         spec = self.spec
         args = ['--enable-c++']
 
+        if spec.satisfies('%clang'):
+            args.append('CFLAGS=-rtlib=compiler-rt')
+
         if '+sigsegv' in spec:
             args.append('--with-libsigsegv-prefix={0}'.format(
                 spec['libsigsegv'].prefix))

